tools/roslaunch/src/roslaunch/loader.py

```diff
@@ -499,6 +499,17 @@ def param_value(self, verbose, name, ptype, value, textfile, binfile, command):
                     # special handle the use of Python scripts in Windows environment:
                     # 1. search for a wrapper executable (of the same name) under the same directory with stat.S_IXUSR flag
                     # 2. if no wrapper is present, prepend command with 'python' executable
+                    def split_with_single_quote_enclosed_fixup(command):
+                        in_command = shlex.split(command, posix=False)  # use non-posix method on Windows
+                        if not ("'" in command):
+                            return in_command
+                        new_command = []
+                        for token in in_command:
+                            if token.startswith("'") and token.endswith("'"):
+                                new_command.append("{}".format(token[1:-1]))
```

Comment:
has this been tested? this might not be a correct fix since `'` are generally used when the string inside contains space(s). When this function return `" ".join(new_command)` (assuming we're still returning a string), that kind of scenarios might not be handled correctly.

I think instead of removing any of the single quotes (I guess they are used because of support from Linux bash), we could try replacing them with `&quot;` (according to https://stackoverflow.com/questions/22468479/are-nested-double-quotes-incorrect-in-xml). I think I used this fix earlier when I was working with another project.

Reply:
Yes, it is tested. I am not really following the scenario here. Why are we using `" ".join(new_command)`?

Reply:
You're right, that would not be necessary